internal/runners/pull/pull.go

```diff
@@ -48,30 +59,49 @@ func (f *outputFormat) MarshalOutput(format output.Format) interface{} {
 	return f
 }
 
-func (p *Pull) Run() error {
+func (p *Pull) Run(params *PullParams) error {
 	if p.project == nil {
 		return locale.NewInputError("err_no_project")
 	}
 
-	if p.project.IsHeadless() {
+	if p.project.IsHeadless() && params.SetProject == "" {
 		return locale.NewInputError("err_pull_headless", "You must first create a project. Please visit {{.V0}} to create your project.", p.project.URL())
 	}
 
-	// Retrieve latest commit ID on platform
-	latestID, err := model.LatestCommitID(p.project.Owner(), p.project.Name())
+	// Determine the project to pull from
+	target, err := targetProject(p.project, params.SetProject)
 	if err != nil {
-		return locale.WrapInputError(err, "err_pull_commit", "Could not retrieve the latest commit for your project.")
+		return errs.Wrap(err, "Unable to determine target project")
+	}
+
+	if params.SetProject != "" {
+		related, err := areCommitsRelated(*target.CommitID, p.project.CommitUUID())
+		if !related && !params.Force {
+			confirmed, err := p.prompt.Confirm(locale.T("confirm"), locale.Tl("confirm_unrelated_pull_set_project", "If you switch to {{.V0}}, you may lose changes to your project.", target.String()), false)
```

Comment:
This is missing the "Are you sure you want to do this?" Otherwise we're just prompting them for yes/no without actually giving them a prompt.

Reply:
Very good point!